gui/wxpython/gis_set.py

```diff
@@ -606,163 +606,90 @@ def ImportFile(self, filePath):
                     'name': filePath},
                 parent=self)
 
+    def CreateMapset(self, event):
+        """Create new mapset"""
+        gisdbase = self.tgisdbase.GetValue()
+        location = self.listOfLocations[self.lblocations.GetSelection()]
+        try:
+            mapset = create_mapset_interactively(self, gisdbase, location)
+            self.OnSelectLocation(None)
+            self.lbmapsets.SetSelection(self.listOfMapsets.index(mapset))
+            self.bstart.SetFocus()
+            return True
+        except Exception as e:
+            GError(parent=self,
+                   message=_("Unable to create new mapset: %s") % e,
+                   showTraceback=False)
+            return False
+
```

Comment:
If you move this to where the old function was in the file, you will get a better view of your changes in the diff. (A more substantial change would be needed to create some better ordering of the functions here, so just keeping what is here seems to be the best course of action.)